Controllers/Recipe_IngredientsController.cs

```diff
@@ -96,15 +96,22 @@ public async Task<JsonResult> UpdateFromAngularController([FromBody]List<View_An
             foreach (View_Angular_Recipe_Ingredients_Details detail in recipeIngredientsDetails)
             {
                 // Dave: why only update if the weight is greater than zero?
-                if (detail.Amount_g > 0)
-                {
-                    var recipeIngredient = await _recitopiaDbContext.Recipe_Ingredients
-                        .SingleAsync(i => i.Id == detail.Id);
+                // Mat: I think I was working through some null or empty value issues. updated
 
-                    recipeIngredient.Amount_g = detail.Amount_g;
+                var recipeIngredient = await _recitopiaDbContext.Recipe_Ingredients
+                        .SingleAsync(i => i.Id == detail.Id);                
 
-                    await _recitopiaDbContext.SaveChangesAsync();
+                if (detail.Amount_g >= 0)
+                {
+                    recipeIngredient.Amount_g = detail.Amount_g;
```

Comment:
Is this to prevent negative values? It's possible to have validation on the submitted values to the controller to prevent this. E.g. the request body or query parameter string

It's also helpful to have front-end validation as the user is inputting.

Reply:
not negative, but blank or zero.  There is some front end validation and will tweak as Vince starts to use and point out some areas of front end improvement.